### PR TITLE
[WIP] Fragmention + Explosion Damage Fixes

### PR DESCRIPTION
**Pull request by JoeKlesczewski**
Going to try to do up a more formal changelog for this:
- [x] Fix problem where shrapnel damage wasn't appropriately getting reduced when a fragment punched through something
- [x] Fix problem where an NPC getting hit by a piece of shrapnel would actually boost its power moving forward
- [ ] Rework fragment power loss to use something other than `player::get_armor_cut`
- [ ] Fix concussive explosive damage ("blast"-type damage) not being reduced by intervening obstacles.
- [x] Fix mixup between `armor_cut_bonus` and `armor_bash_bonus`
- [X] Fix oversight where fire-type explosions weren't having their relative noise reduction applied
- [X] Changed explosion damage to fall off via inverse square of distance, rather than linearly
- [X] Fix zombies/NPCs preventing the player from taking any damage from shrapnel if they are on the same tile
- [X] Shrapnel now damages vehicles properly

This is related to issue #10835

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -5653,17 +5651,12 @@ void game::draw_critter(const Creature &critter, const point &center)
...
         critter.draw( w_terrain, center.x, center.y, false );
         return;
     }
-    const bool has_ir = u.has_active_bionic( "bio_infrared" ) ||
-                        u.has_trait( "INFRARED" ) ||
-                        u.has_trait( "LIZ_IR" ) ||
-                        u.worn_with_flag( "IR_EFFECT" );
-    const bool can_see = m.pl_sees( u.posx, u.posy, critter.xpos(), critter.ypos(),
-                                    u.sight_range( DAYLIGHT_LEVEL ) );
-    if( critter.is_warm() && has_ir && can_see ) {
```
Comment (KA101) [inline src/game.cpp]: Uh, we removed this a day or two ago.  Might wanna `git pull upstream master` and make sure your master branch is up-to-date.  (Test in a new branch first.  Most of the time it works in place of `git rebase` but when it doesn't, the PR's unmergeable.)

Comment (QuantumInvictus) [inline src/game.cpp]: Interesting. I'd thought I did that, but apparently not.

**Comment by kevingranade** [comment]
hah, that second one's a doozy.

**Comment by QuantumInvictus** [comment]
Okay, currently I have set it up such that a piece of furniture will reduce blast damage by half of its minimum strength for bashing. This doesn't seem to be a very good way of handling it, but I can't think of any other way of doing it. Does anyone have any suggestions? Perhaps averaging the min and max strength values...

**Comment by KA101** [comment]
Hey, what's the story on this one?

**Comment by JoeKlesczewski** [comment]
First up, my apologies for dropping off the face of the earth there - life got in the way and I wound up losing the password to this account. By blind luck I found it again when I went to do some stuff for college, and I'm only just now getting back to this.

With that out of the way, it's pretty clear that this branch is quite a bit behind the current state of the code. As such, I figure that I might as well start over from scratch and reimplement the things I did, as that would probably be easier all around. Which leads me to a question - is it possible to replace one branch with another as the pull request's relevant branch?

**Pull request closed without merging**